server/analytics/reaction.go

```diff
@@ -246,6 +246,53 @@ func (a Analytics) GetReactionCount(arg GetReactionCountParams) (int, error) {
 	return count, row.Scan(&count)
 }
 
+const getReactionCountFiltered = `
+SELECT count(*)
+FROM reactions
+WHERE organization_id = ?
+AND action_id = ?
+%s
+%s
+AND created_at BETWEEN ? AND ?
+`
+
+type GetReactionCountFilteredParams struct {
+	OrganizationID uuid.UUID
+	ActionID       uuid.UUID
+	EventIDs       []string
+	UserID         string
+	Start          time.Time
+	End            time.Time
+}
+
+func (a Analytics) GetReactionCountFiltered(arg GetReactionCountFilteredParams) (int, error) {
+	var count int
+
+	var identity = ""
+	var event = ""
+	if arg.UserID != "" {
+		identity = fmt.Sprintf(
+			"AND identity_id = '%s'",
```

Comment:
This can be dangerous as you could inject malicious SQL. Please always escape the params. But I would suggest to use `?` instead of `%s` and append the value of `%s` to a params slice which you later supply to the `QueryRowx()` method like `QueryRowx(query, params)`.

`params` can initially be:

```
params := []string{
    arg.OrganizationID,
    arg.ActionID,
    arg.Start,
    arg.End,
}
```

Reply:
function removed